### ISSUE TITLE
Tedit created world persistant visual bug, tiles not visible

### ISSUE THREAD

**Issue by allelsewastaken** [TEdit/Terraria-Map-Editor#894]
I created a new world in Tedit, with the ground/cavern levels set so the world was entirely empty. no tiles or walls, at all. I then edited the world so there was a crimson desert covering the bottom of the world.

When I attempted to play the world only the surface 10 or so blocks of sand are visible on screen. Beneath that it just shows a massive empty area, as if there were no tiles there at all. When I use weapons that illuminate distant tiles the missing blocks show up for a second then immediately disappear again.

I have tried new characters and further world editing but nothing fixes the bug. Please see screenshot:

http://imgur.com/a/ZD3wW

That is entirely sand, with nothing holding it up, and nothing I try changes that those blocks are not displayed.

**Comment by enigmaquip** [comment]
It's not really a TEdit bug, at best it's a complaint at Tedit not completely babysitting you on changes. Setting the ground/cavern levels outside their normal bounds causes this

**Comment by enigmaquip** [comment]
http://forums.terraria.org/index.php?threads/tedit-terraria-map-editor.43822/page-2#post-1008532

**Comment by allelsewastaken** [comment]
So TEdit simply lets you make a change that the game rendering engine does not know how to deal with. Annoying, but fair. Thanks. :)

**Comment by enigmaquip** [comment]
The response is spot on, love it. Yea I could probably do a level check when saving and warn if it's outside normal bounds....

**Comment by allelsewastaken** [comment]
I have been doing some experimenting. I created a few new worlds in TEdit of different sizes and played with changing the surface and cavern layers. I found that the disappearing tile issue seems to have nothing to do with the surface/cavern settings. Rather it is the altitude of tiles you are standing on. I am not sure how to explain what I found to you.

In my testing I found that at some point a bit beneath the midpoint of any map any tiles not currently illuminated are simply not rendered. This happens in every Terraria world, but we can only see it in worlds where there is no background wall to keep 'sunlight' from showing through where the tiles should be. When the surface/cavern levels are high enough the default background fulfills this purpose.

I speculate that this serves to help speed up the game and reduce resource usage. It makes sense, at least to me. Under normal circumstances those 'missing' tiles would not be missed. And if a source of light ever IS shown upon them they suddenly show up again. This is observably the current behavior.

So the whole thing is not a bug or oversight, but may be a design feature to improve game performance. Thoughts? :)
